rclpy/CMakeLists.txt

```diff
@@ -146,13 +194,24 @@ target_link_libraries(_rclpy_handle PRIVATE
   rcutils::rcutils
 )
 configure_build_install_location(_rclpy_handle)
+clean_windows_flags(_rclpy_handle)
 
 if(NOT WIN32)
   ament_environment_hooks(
     "${ament_cmake_package_templates_ENVIRONMENT_HOOK_LIBRARY_PATH}"
   )
 endif()
 
+# Workaround to ensure rclpy_common has same definition for Py_DEBUG
+# TODO(sloretz) remove this workaround when using pybind11 2.6+
+get_target_property(_rclpy_compile_definitions _rclpy COMPILE_DEFINITIONS)
+foreach(_rclpy_definition ${_rclpy_compile_definitions})
+  if("Py_DEBUG" STREQUAL _rclpy_definition)
+    target_compile_definitions(rclpy_common PUBLIC Py_DEBUG)
+    break()
+  endif()
+endforeach()
```

Comment:
Given what we now know, it seems like this isn't the correct workaround.  That is, we specifically *don't* want to add `Py_DEBUG` to the command-line, since that ends up with the warning.  It kind of looks to me like we need to remove this, and then see if there is something else that is also adding `Py_DEBUG` to the command-line.  If it is pybind11, maybe the thing to do here is to augment `clean_windows_flags` to also remove that flag?